packages/widget/src/components/SelectChainAndToken.tsx

```diff
@@ -36,6 +39,16 @@ export const SelectChainAndToken: React.FC<BoxProps> = (props) => {
     toToken &&
     !prefersNarrowView &&
     !hiddenToToken;
+
+  useEffect(() => {
+    emitter.emit(WidgetEvent.ExecutionPathDetails, {
+      fromChain,
+      fromToken,
+      toChain,
+      toToken,
+    });
+  }, [fromChain, toChain, fromToken, toToken]);
```

Comment:
We should emit events when we select the token, not inside useEffect. This will emit unnecessary events when mounted.